server/internal/usecase/interactor/schema.go

```diff
@@ -62,40 +63,245 @@ func (i Schema) CreateField(ctx context.Context, param interfaces.CreateFieldPar
+		var res *schema.Field
+
+		f1.TypeProperty().Match(schema.TypePropertyMatch{
+			Reference: func(f *schema.FieldReference) {
+				fr = f
+			},
+		})
+
+		if fr.CorrespondingField() != nil {
+			cf1 := fr.CorrespondingField()
+			mId2 := fr.Model()
+			m2, err := i.repos.Model.FindByID(ctx, mId2)
+			if err != nil {
+				return err
+			}
+			s2, err := i.repos.Schema.FindByID(ctx, m2.Schema())
+			if err != nil {
+				return err
+			}
+
+			if !operator.IsMaintainingProject(s2.Project()) {
+				return interfaces.ErrOperationDenied
+			}
+
+			if cf1.Key == nil || s2.HasFieldByKey(lo.FromPtr(cf1.Key)) {
+				return schema.ErrInvalidKey
+			}
+
+			// set f2 type property, modelId = mId1, cf2 = f1
+			cf2 := &schema.CorrespondingField{
+				FieldID:     f1.ID().Ref(),
+				Title:       lo.ToPtr(f1.Name()),
+				Key:         lo.ToPtr(f1.Key().String()),
+				Description: lo.ToPtr(f1.Description()),
+				Required:    lo.ToPtr(f1.Required()),
+			}
+			tp := schema.NewReference(param.ModelId, cf2, cf2.FieldID).TypeProperty()
+
+			// create f2 from cf1
+			f2, err := schema.NewField(tp).
+				NewID().
+				Unique(false).
+				Multiple(false).
+				Required(lo.FromPtr(cf1.Required)).
+				Name(lo.FromPtr(cf1.Title)).
+				Description(lo.FromPtr(cf1.Description)).
+				Key(key.New(lo.FromPtr(cf1.Key))).
+				DefaultValue(nil).
+				Build()
+			if err != nil {
+				return err
+			}
+
+			s2.AddField(f2)
+
+			if err := i.repos.Schema.Save(ctx, s2); err != nil {
+				return err
+			}
+
+			res = f2
+		}
+
+		// set corresponding field
+		fr.SetCorrespondingField(res.ID().Ref())
+
+		return nil
 	})
 }
 
 func (i Schema) UpdateField(ctx context.Context, param interfaces.UpdateFieldParam, operator *usecase.Operator) (*schema.Field, error) {
 	return Run1(ctx, operator, i.repos, Usecase().Transaction(), func(ctx context.Context) (*schema.Field, error) {
-		s, err := i.repos.Schema.FindByID(ctx, param.SchemaId)
+		s1, err := i.repos.Schema.FindByID(ctx, param.SchemaId)
 		if err != nil {
 			return nil, err
 		}
 
-		if !operator.IsMaintainingProject(s.Project()) {
+		if !operator.IsMaintainingProject(s1.Project()) {
 			return nil, interfaces.ErrOperationDenied
 		}
 
-		f := s.Field(param.FieldId)
-		if f == nil {
+		f1 := s1.Field(param.FieldId)
+		if f1 == nil {
 			return nil, interfaces.ErrFieldNotFound
 		}
 
-		if err := updateField(param, f); err != nil {
+		// check if type is reference
+		if f1.Type() == value.TypeReference {
+			err := i.updateCorrespondingField(ctx, f1, param, operator)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		if err := updateField(param, f1); err != nil {
 			return nil, err
 		}
-		if err := i.repos.Schema.Save(ctx, s); err != nil {
+		if err := i.repos.Schema.Save(ctx, s1); err != nil {
 			return nil, err
 		}
 
-		return f, nil
+		return f1, nil
+	})
+}
+
+func (i Schema) updateCorrespondingField(ctx context.Context, f1 *schema.Field, param interfaces.UpdateFieldParam, operator *usecase.Operator) error {
```

Comment:
This func is actually domain logic, because there is few logics to load and save something with repos in the function.

```go
// pkg/schema/corresponding_field.go

type CorrespondingFieldTuple struct {
	Schema1 *Schema
	Field1  *Field
	Schema2 *Schema
	Field2  *Field
}

func GetOrCreateCorrespondingFields(s1, s2 *Schema, fid FieldID) (CorrespondingFieldTuple, error) {
	// ...
}
```